Lib/test/libregrtest/setup.py

```diff
@@ -14,17 +14,27 @@
 
 
 def setup_tests(ns):
-    # Display the Python traceback on fatal errors (e.g. segfault)
-    faulthandler.enable(all_threads=True)
-
-    # Display the Python traceback on SIGALRM or SIGUSR1 signal
-    signals = []
-    if hasattr(signal, 'SIGALRM'):
-        signals.append(signal.SIGALRM)
-    if hasattr(signal, 'SIGUSR1'):
-        signals.append(signal.SIGUSR1)
-    for signum in signals:
-        faulthandler.register(signum, chain=True)
+    try:
+        stderr_fd = sys.stderr.fileno()
+    except ValueError:
+        # For example, in IDLE, sys.stderr has no file descriptor,
+        # so faulthandler cannot be used.
+        #
+        # Catch ValueError to catch io.UnsupportedOperation on TextIOBase
+        # and ValueError on a closed stream.
+        stderr_fd = None
```

Comment:
Replace line with
```
        try:
            stderr_fd = sys.__stderr__.fileno()
        except ValueError:
            stderr_fd = None
```
If IDLE is started from a command line or python (by import idlelib.idle), the backup call returns 2.
>>> sys.__stderr__
<_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>
>>> sys.__stderr__.fileno()
2

Reply:
done